datafusion/common/src/column.rs

```diff
@@ -382,6 +382,7 @@ mod tests {
     use arrow::datatypes::{DataType, SchemaBuilder};
     use std::sync::Arc;
 
+    #[allow(clippy::needless_pass_by_value)]
     fn create_qualified_schema(qualifier: &str, names: Vec<&str>) -> Result<DFSchema> {
```

Comment:
we could change this to `impl IntoIterator<Item = &str>` for example

```suggestion
    fn create_qualified_schema(qualifier: &str, names: impl IntoIterator<Item = &str>) -> Result<DFSchema> {
```